components/scream/src/physics/p3/tests/p3_run_and_cmp.cpp

```diff
@@ -15,51 +15,86 @@ using namespace scream;
+   * runs the new/experimental version of the code and compares it against
+   * the baseline data you've saved to file. Both baseline and cmp modes
+   * start from an initial condition in ../p3_ic_cases.cpp. By default, 
+   * tests are run with log_PredictNc=true and false and also for 1 step or
+   * 6 steps. This creates a total of 4 different cases. When 6 steps are run,
+   * output for each step is considered separately. 
+   */
+
+
+/* Given a column of data for variable "label" from the reference run 
+ * (probably master) and from your new exploratory run, loop over all 
+ * heights and confirm whether or not the relative difference between  
+ * runs is within tolerance "tol". If not, print debug info. Here, "a"
+ * is the value from the reference run and "b" is from the new run.
+ */  
 template <typename Scalar>
 static Int compare (const std::string& label, const Scalar* a,
                     const Scalar* b, const Int& n, const Real& tol) {
-  Int nerr = 0;
+
+  Int nerr1 = 0;
+  Int nerr2 = 0;
   Real den = 0;
   for (Int i = 0; i < n; ++i)
     den = std::max(den, std::abs(a[i]));
   Real worst = 0;
   for (Int i = 0; i < n; ++i) {
     if (std::isnan(a[i]) || std::isinf(a[i]) ||
         std::isnan(b[i]) || std::isinf(b[i])) {
-      ++nerr;
+      ++nerr1;
       continue;
     }
+    
     const auto num = std::abs(a[i] - b[i]);
     if (num > tol*den) {
-      ++nerr;
+      ++nerr2;
       worst = std::max(worst, num);
     }
   }
-  if (nerr)
-    std::cout << label << " nerr " << nerr << " worst " << (worst/den)
-              << " with denominator " << den << "\n";
-  return nerr;
+
+  if (nerr1) {
+    std::cout << label << " has " << nerr1 << " infs + nans.\n";
+
+  }
+  
+  if (nerr2) {
+    std::cout << label << " > tol " << nerr2 << " times. Max rel diff= " << (worst/den)
+	      << " normalized by ref impl val=" << den << ".\n";
+
+  }
+  
+  return nerr1 + nerr2;
 }
 
-Int compare (const std::string& label, const double& tol,
+ /* When called with the below 3 args, compare loops over all variables 
+  * and calls the above version of "compare" to check for and report 
+  * large discrepancies.
+  */
+ Int compare (const double& tol,
              const FortranData::Ptr& ref, const FortranData::Ptr& d) {
+ 
   Int nerr = 0;
   FortranDataIterator refi(ref), di(d);
   scream_assert(refi.nfield() == di.nfield());
   for (Int i = 0, n = refi.nfield(); i < n; ++i) {
     const auto& fr = refi.getfield(i);
     const auto& fd = di.getfield(i);
     scream_assert(fr.size == fd.size);
-    nerr += compare(label + std::string(" ") + fr.name,
-                    fr.data, fd.data, fr.size, tol);
+    nerr += compare(fr.name, fr.data, fd.data, fr.size, tol);
   }
   return nerr;
 }
 
 struct Baseline {
   Baseline () {
     for (const bool log_predictNc : {true, false})
-      for (const int it : {1, 6})
+      for (const int it : {1, 6})   // # of steps to run is either 1 or 6.
```

Comment:
@ambrad - I don't understand why we'd ever want to run for multiple different lengths of time since the "run 6 steps" case runs and reports the result after 1 step as well. I checked empirically and the 1 step results are identical to the first step of the 6th step results... Can I delete the looping over nsteps here?